appinventor/components/src/com/google/appinventor/components/runtime/ImagePicker.java

```diff
@@ -102,114 +73,10 @@ protected Intent getIntent() {
    */
   public void resultReturned(int requestCode, int resultCode, Intent data) {
     if (requestCode == this.requestCode && resultCode == Activity.RESULT_OK) {
-      Uri selectedImage = data.getData();
-      selectionURI = selectedImage.toString();
-      Log.i(LOG_TAG, "selectionURI = " + selectionURI);
-
-      // get the file type extension from the intent data Uri
-      ContentResolver cR = container.$context().getContentResolver();
-      MimeTypeMap mime = MimeTypeMap.getSingleton();
-      String extension = "." + mime.getExtensionFromMimeType(cR.getType(selectedImage));
-      Log.i(LOG_TAG, "extension = " + extension);
-
-      // save the image to a temp file in external storage, using a name
-      // that includes the extension
-      saveSelectedImageToExternalStorage(extension);
+      selectedImage = data.getDataString();
```

Comment:
This code code does not work for me.   Did you test it?    I wrote a simple program where AfterPicking sets the screen  background to the result returned by the picker and it does not work.    The issue here is that data.getDataString() is a content URL and you need to return an actual file path.   The code in the version of resultReturned your removed was doing that conversion by writing the result to a file in external storage.   That seems like a waste, but I don't know a better way to do it, and I didn't see a better method on stackoverflow.   Maybe you can find it.  I might have missed something.

Let me know if you want to just give up on fixing this.  But first look arond